### PR TITLE
198 change goal to workspace

### PR DESCRIPTION
**Pull request by JamesVetro**
This pull request IS NOT FINISHED. This request needs to be worked on by front end in order to make this work. However, the back end section of this is done, and so it is ready to be pushed to their side.

### CODE REVIEW

**Review by cmarcusdimarco — CHANGES_REQUESTED** [review]
All of the goal -> workspace refactoring looks good at first glance, but this pull request has some additional changes that are not related to the renaming. Please revert the changes to the resource and topic controllers so we can localize those changes to their related issue.

File: server/controller/apis/resourceController.js [hunk 1]
```diff
@@ -289,17 +289,19 @@ exports.saveResource = async ( req, res, redirect ) => {
         }
         else if ( !req.files || Object.keys( req.files ).length === 0 ) {   // no files were uploaded       
             // no files uploaded
-            if( resource.resourceType == 1 ) {
-                this.saveResourceImage( req, res, resource.id, 'notebook-pen.svg' );
-            }
-            else if ( resource.resourceType == 2 ) {
-                this.saveResourceImage( req, res, resource.id, 'cell-molecule.svg' );
```
Comment (Mike-Volchko) [inline server/controller/apis/resourceController.js:296]: As Marcus suggested. We need to revert these save image changes in resources, topics, and workspaces. They do not relate to this ticket and use the nested if structure that we talked about removing on 10/24.

**Review by Mike-Volchko — CHANGES_REQUESTED** [review]
Please revert the save image changes in all 3 apis.

**Review by briangormanly — CHANGES_REQUESTED** [review]
Looks like this one might not make the cut for the lint changes.. There are also so conflicts now that other PR's have been merged, make sure to pull main back into this branch / resolve the conflicts before resubmitting.

**Comment by briangormanly** [comment]
Make sure to pull main into this branch! there are a lot of conflicts that need to be taken care off.

**Comment by JamesVetro** [comment]
All the changes

**Comment by JamesVetro** [comment]
The duplication here is to be closed, no further work on this PR needed

**Pull request closed without merging**